File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    // Kadane’s Algorithm
    let msum=nums[0];
    let sum=nums[0];
    for(let i=1;i<nums.length;i++){
        sum=Math.max(nums[i],sum+nums[i])
        msum=Math.max(msum,sum);
    }
    return msum;
};

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int msum=nums[0];
        int sum=nums[0];
        for(int i=1;i<nums.size();i++){
            sum=max(nums[i],sum+nums[i]);
            msum=max(msum,sum);
        }
        return msum;
    }
};